### DISCUSSION TITLE
(newbie question) Budibase - private messaging system

### DISCUSSION THREAD

**Discussion by KB-WEB-DEVELOPMENT** [Budibase/budibase#5691] [Help]
Hello,

(question from a newbie)

I have been looking for a while now for a tool to build a private messaging system  within a web application.

Could I use Budibase in the following way :

Table name #1 : new_message_table

fields :

- id (primary key)
- sender_id (foreign key, i.e :  relationship to users table, user_id field)
- recipient_id (foreign key, i.e :  relationship to users table, user_id field)
- message_read (boolean)
- message_title
- message_content
( +  common standard fields such as created_at, etc ...)

Table name #2 : reply_to_message_table

fields :

- id (primary key)
- sender_id (foreign key, i.e :  relationship to users table, user_id field)
- recipient_id (foreign key, i.e :  relationship to users table, user_id field)
- message_read (boolean)
- message_title (foreign key, i.e :  relationship to new_message_table, message_title field)
- message_content
( +  common standard fields such as created_at, etc ...)

I am still learning about Budibase automation but the idea would be that once a new message or reply message has been sent,
the recipient would receive an email notification in his private (gmail, yahoo, etc ...) inbox.

Is this the right approach or do you see it in a completely different way  ?

Thank you very much for your time and patience.

Regards,

KBW

**Comment by cverbiest** [comment]
On the risk of being off-topic, don`t build your own messaging system. 
pick an existing one that fits your requirements.

**Comment by melohagan** [comment]
Hey @KB-WEB-DEVELOPMENT 

I think @cverbiest's advice is probably wise.

Budibase does have integrations with other existing messaging apps, including email:

- https://docs.budibase.com/docs/email
- https://docs.budibase.com/docs/slack-notification
- https://docs.budibase.com/docs/discord

If you want to send an email on row creation that would be fairly easy. 

1. Configure the SMTP settings 
2. Add an automation with a row creation trigger: https://docs.budibase.com/docs/trigger
3. Add the send email action.

**Comment by KB-WEB-DEVELOPMENT** [comment]
@cverbiest & @melohagan  - thanks a lot. KBW .
